### PR TITLE
Das/pytutorial 3

### PR DESCRIPTION
**Pull request by mezarque**
First draft of Lesson 3! I've been reflecting on some feedback and I think it would be nice for us to show some demos of ways in which people can use Python for their work.

Unfortunately I'll be out for the whole time between lesson 2 and 3 (Mar 14-25) so I won't be able to prepare any new materials. But perhaps other folks could prepare some examples of things they use Python to do that are better than done manually?

I'm also thinking we could solicit some questions from folks at the end of Lesson 2, basically asking "What would you like to know how to do using Python?" or "What's a computing task you wish you could automate?".

### CODE REVIEW

File: docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md [hunk 1]
```diff
@@ -0,0 +1,1123 @@
...
+
+
+    CompletedProcess(args=['which', 'python'], returncode=0)
+
+
+
+The code above runs the same as `!which python`.  
+`subprocess.run` passes each element of a list as an argument to the shell. 
+
+It's unfortunately not as tidy looking, but it does allow for more robust code that can be converted to Python packages or functions without rewriting!
```
Comment (taylorreiter) [inline docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md:192]:
```suggestion
It's unfortunately not as tidy looking, but it does allow for more robust code that can be converted to Python scripts without rewriting!
```

File: docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md [hunk 2]
```diff
@@ -0,0 +1,1123 @@
...
+    
+![png](lesson_files/lesson_52_0.png)
+    
+
+
+You'll notice that rather than simply using `import matplotlib.pyplot`, we added an `as plt` expression to the end of our import statement.  
+Python allows you to import packages using aliases.  
+This saves you from having to type out `matplotlib.pyplot.plot()` and shortens the expression to `plt.plot()`.  
+
+Package documentation examples often gives you a suggestion for how you might abbreviate packages for others to be able to understand.
```
Comment (taylorreiter) [inline docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md:561]:
```suggestion
Package documentation examples often give you a suggestion for how you might abbreviate packages for others to be able to understand.  
```

File: docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md [hunk 3]
```diff
@@ -0,0 +1,1123 @@
...
+```
+
+
+    
+![png](lesson_files/lesson_54_0.png)
+    
+
+
+This example only scratches the barest surface of what you can do with **`plt`**.  
+Future workshops will dive into deeper detail about how **`matplotlib`** works and what you can do with it.
```
Comment (taylorreiter) [inline docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md:616]:
```suggestion
If people want to learn more, we can have future workshops that dive into deeper detail about how **`matplotlib`** works and what you can do with it.
```

Comment (taylorreiter) [reply docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md:616]: you can keep it phrased as is, but I don't know enough about matplotlib to teach more about it and I didn't want you to feel locked in to covering it!

File: docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md [hunk 4]
```diff
@@ -0,0 +1,1123 @@
...
+### 3.3 Working with `DataFrame`s using **`pandas`**
+
+A popular package for dealing with tabular data like `.tsv` and `.csv` files is **`pandas`**.  
+You can use **`pandas`** to load files and perform all manner of database manipulations using `DataFrame` data structures.
+
+
+```python
+import pandas as pd
+
+df = pd.DataFrame(pd.read_csv('cities.csv'))
```
Comment (taylorreiter) [inline docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md:628]: does this have to be wrapped in `pd.DataFrame()`? I've never seen that before

File: docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md [hunk 5]
```diff
@@ -0,0 +1,1123 @@
...
+
+As you might notice, **`pandas`** can do rather complex operations using relatively few commands.  
+If you tried to manually build some of these functions, it could be quite challenging!  
+
+Hopefully these examples give you a sense of what you can do using Python if you build your skills and become fluent in the language.
+
+
+```python
+
+```
```
Comment (taylorreiter) [inline docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md:1123]: you can leave this if you want but we could also delete it
```suggestion
```

**Review by taylorreiter — COMMENTED** [review]
Do you have a folder on your computer called `lesson_files`, and that has files like `lesson_files/lesson_52_0.png`? The pngs don't render in `lesson.md` for me because those files weren't a part of the PR, can you add them? (`lesson_52_0.png`, `lesson_54_0.png`, are the ones I caught as missing).

As to your comments in the PR, I love the idea of soliciting use cases from people. I think we should definitely do that. As far as a show and tell for uses in python, this lesson is pretty long so I'm not sure we'll have time for that. we could definitely fit that type of thing in at a later AUG!

File: docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md [hunk 6]
```diff
@@ -0,0 +1,1123 @@
...
+<b>import</b> os
+
+numbers = [1, 2, 3, 4, 5]
+
+destination = 'testfiles/'
+os.mkdir(destination)
+
+<b>for</b> number <b>in</b> numbers:
+    filename = 'file' + str(number) + '.txt'
+    subprocess.run(['touch', destination + filename])
```
[the remark this answers is not in the document]
Comment (mezarque) [reply docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md:256]: I think the more robust/rigorous way to do it definitely us to use os.path! I'll add it as another example answer.

**Pull request merged**